models/slotsModel.js

```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+
+const slotSchema = new mongoose.Schema({
+  doctorid: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Doctor',
+    required: [true, 'a doctor is required'],
+  },
+  clinicid: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Clinic',
+    required: [true, 'a clinic is required'],
+  },
+  timeSlot: {
+    type: Date,
+    required: [true, 'a slot is required'],
+  },
+  duration: {
```

Comment:
this also should have a validator for the duration to be integers greater than 0